packages/synapse-react-client/src/components/QueryVisualizationWrapper/QueryVisualizationWrapper.tsx

```diff
@@ -117,18 +120,19 @@ export function QueryVisualizationWrapper(
 ) {
   const {
     noContentPlaceholderType = NoContentPlaceholderType.INTERACTIVE,
-    columnAliases = {},
     defaultShowSearchBar = false,
     defaultShowFacetVisualization = true,
     unitDescription = 'result',
   } = props
 
-  const {
-    data,
-    getCurrentQueryRequest,
-    isFacetsAvailable,
-    hasResettableFilters,
-  } = useQueryContext()
+  const columnAliases = useMemo(
+    () => props.columnAliases ?? {},
+    [props.columnAliases],
+  )
```

Comment:
The previous initializer (default value `{}`) was re-created every render, which triggered undesirable rerenders.